src/HotChocolate/Core/src/Execution/Batching/BatchExecutor.Enumerable.cs

```diff
@@ -46,9 +50,80 @@ private class BatchExecutorEnumerable : IAsyncEnumerable<IQueryResult>
             _inputFormatter = inputFormatter ??
                 throw new ArgumentNullException(nameof(inputFormatter));
             _visitor = new CollectVariablesVisitor(requestExecutor.Schema);
+            _allowParallelExecution = allowParallelExecution;
         }
 
-        public async IAsyncEnumerator<IQueryResult> GetAsyncEnumerator(
+        public IAsyncEnumerator<IQueryResult> GetAsyncEnumerator(
+            CancellationToken cancellationToken = default)
+        {
+            if (_allowParallelExecution)
+            {
+                return GetParallelAsyncEnumerator(cancellationToken);
+            }
+            else
+            {
+                return GetSequentialAsyncEnumerator(cancellationToken);
+            }
+        }
+
+        private async IAsyncEnumerator<IQueryResult> GetParallelAsyncEnumerator(
+            CancellationToken cancellationToken = default)
+        {
+            var pending = new List<Task<IQueryResult>>();
+            var exportCount = _exportedVariables.Count;
+            var hasMutatedState = false;
+            BatchScheduler? scheduler = null;
+            foreach (IQueryRequest queryRequest in _requestBatch)
+            {
+                var nextScheduler = queryRequest.Services.GetRequiredService<IBatchDispatcher>() as BatchScheduler;
```

Comment:
This is probably not the correct way to solve the problem, but what I'm trying to do here is to avoid that the WorkScheduler.ExecuteAsync forces the dispatch of the current queue when it ends. We want to be able to register all outbound tasks before we allow it to dispatch, so that we can get them in the same batch